components/Connect.tsx

```diff
@@ -70,6 +84,42 @@ export default function BluetoothConnect({ bikeCredentials, setBikeInstance, bac
                     Back to login
                 </Button>
             </P>
+
+            {/* If Backup button pressed hide it */}
+            {!ShowBackupKey && (
+                <P vertical={10}>
+                    <Button
+                        onClick={() => setShowBackupKey(true)}
+                        secondary
+                    >
+                        Backup login credentials
+                    </Button>
+                </P>
+            )}
+
+            {/* If Backup button pressed show the key */}
+            {ShowBackupKey && (
+                <>
+                    {/* Copy button */}
+                    <Button onClick={handleCopy} secondary>
+                        Copy credentials to clipboard
+                    </Button>
+
+                    <P vertical={10}>
+                        {/* Download button */}
+                        <Button onClick={handleDownload} secondary>
+                            Download credentials as file
+                        </Button>
+                    </P>
+
+                    {/* Show the key */}
+                    <textarea
+                        readOnly
+                        value={parsedCredentials}
+                        style={{ width: '85%', height: '200px', fontSize: '0.5rem' }}
+                    />
+                </>
+            )}
```

Comment:
This can become an inline if else

```suggestion
            {ShowBackupKey ? (
                <div>
                    {/* If Backup button pressed show the key */}

                    <Button onClick={handleCopy} secondary>
                        Copy credentials to clipboard
                    </Button>

                    <P vertical={10}>
                        <Button onClick={handleDownload} secondary>
                            Download credentials as file
                        </Button>
                    </P>

                    <textarea
                        readOnly
                        value={parsedCredentials}
                        style={{ width: '85%', height: '200px', fontSize: '0.5rem' }}
                    />
                </div>
            ) : (
                <P vertical={10}>
                    {/* If Backup button pressed hide it */}

                    <Button
                        onClick={() => setShowBackupKey(true)}
                        secondary
                    >
                        Backup login credentials
                    </Button>
                </P>
            )}
```